src/argilla/client/feedback/training/schemas.py

```diff
@@ -973,12 +982,95 @@ def _prepare_for_training_with_trl(
         return ds
 
 
+class TrainingTaskForChatCompletionFormat(BaseModel):
+    """
+    Union[Tuple[str, str, str, str], List[Tuple[str, str, str, str]]]
+    """
+
+    format: Union[Tuple[str, str, str, str], List[Tuple[str, str, str, str]]]
+
+
+class TrainingTaskForChatCompletion(BaseModel, TrainingData):
+    _formatting_func_return_types = TrainingTaskForChatCompletionFormat
+    formatting_func: Callable[[Dict[str, Any]], Union[None, str, Iterator[str]]]
+
+    def _format_data(self, dataset: "FeedbackDataset") -> List[Dict[str, str]]:
+        output = set()
+        for sample in dataset.format_as("datasets"):
+            chat_turn_role_content = self.formatting_func(sample)
+            if chat_turn_role_content is None:
+                continue
+
+            self._test_output_formatting_func(chat_turn_role_content)
+
+            if isinstance(chat_turn_role_content, tuple):
+                chat_turn_role_content = {chat_turn_role_content}
+
+            output |= set(chat_turn_role_content)
+        return [{"chat": chat, "turn": turn, "role": role, "content": content} for chat, turn, role, content in output]
+
+    @property
+    def supported_frameworks(self):
+        names = ["openai"]
```

Comment:
This should probably not be here

Reply:
This is intended because we only support OpenAI for this task